add compileInsertGetId and processInsertGetId method

donnykurnia opened the pull request:
taken from PostgresGrammar and PostgresProcessor classes as-is

I've check it in my project using php artisan tinker and it's working well
PS: this library need tests to make sure all methods working well

Signed-off-by: Donny Kurnia donnykurnia@gmail.com

donnykurnia committed:
add compileInsertGetId and processInsertGetId method taken from PostgresGrammar and PostgresProcessor classes as-is
I've check it in my project using php artisan tinker and it's working well
PS: this library need tests to make sure all methods working well

Signed-off-by: Donny Kurnia <donnykurnia@gmail.com>

donnykurnia commented:
PS: This approach assume that the tables already have generator and trigger set for the id column.

cedamorim commented:
nice approach, assuming that the tables that are using generator / sequence will be with trigger, I think this solves our problems

donnykurnia committed:
add processColumnListing Processor

donnykurnia committed:
add

donnykurnia committed:
trim the column name

donnykurnia committed:
handle pdo_firebase errors "already an active transaction"
Reference:
https://secure.php.net/manual/en/pdo.begintransaction.php#114788
https://stackoverflow.com/questions/19216375/transaction-not-working/19220044#19220044

donnykurnia committed:
additional checking before set PDO::ATTR_AUTOCOMMIT attribute

jessesantos commented:
It works for Laravel 5.2?

donnykurnia commented:
@jessesantos my fork only tested for laravel 5.1, because my project still use it for now.

johanWP commented:
Does it work with PHP7?
